### ISSUE TITLE
DenseLayer.ForwardGpu returns all-zero output when activationFunction is null (linear layer) — exposed by #1625 fix

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1629]
## Summary
After the #1625 fix landed (PR #1628), `Predict` returns real probabilities for binary classifiers (Sigmoid head) and multi-class (Softmax). However a separate latent bug surfaces for **regression models with linear output (`activationFunction: null` on the last Dense layer)**: their GPU forward path returns exact 0.0 for every sample.

## Repro
Facade walkthrough Example 2 (Sales Forecasting, Regression task type, MSE loss, last layer `new DenseLayer<double>(outputSize: 1)` with no activation):

```
AIDOTNET_DISABLE_GPU=0 → pred=(0.000000, 0.000000, 0.000000, 0.000000, 0.000000)  exactZeros=15/15  RMSE=0.685
AIDOTNET_DISABLE_GPU=1 → pred=(1.094921, 0.116904, 1.014596, 0.213630, 1.134383)  exactZeros=0/15   RMSE=0.128
```

Same code, same data, same model, same training. The CPU path produces a well-trained regressor (RMSE 0.128); the GPU path produces literal-zero output for every test sample.

Adding `activationFunction: new SigmoidActivation<double>()` to the same last layer also makes the GPU path produce non-zero output. The trigger is specifically `activationFunction: null` (resolving to `FusedActivationType.None`).

## Suspected root cause
`DenseLayer.ForwardGpu` (DenseLayer.cs ~line 1094) calls:
```csharp
var result = gpuEngine.FusedLinearGpu(input2D, _weights, _biases, fusedActivation);
```

When `fusedActivation == FusedActivationType.None` (linear path), `FusedLinearGpu` appears to return a tensor of zeros instead of `input @ W + b`. `FusedLinearGpu` lives in the AiDotNet.Tensors NuGet package so I couldn't trace deeper from the consumer side, but the evidence is:
- Same call path with any non-None activation works correctly post-#1625-fix.
- CPU path (`DenseLayer.Forward`, which doesn't call `FusedLinearGpu`) works correctly.
- The output tensor is literally `== 0.0` (verified via `:F6` print + exact-equality count), not "very small values".

## Impact
- All regression models built through the facade with a linear output head produce zero predictions on GPU-equipped boxes.
- Likely also affects any model with intermediate linear (no-activation) Dense layers, e.g. pre-activation projections in custom architectures.

## Suggested investigation
1. Check `DirectGpuTensorEngine.FusedLinearGpu`'s code path for `FusedActivationType.None`.
2. If the None branch is simply forwarding the pre-activation tensor's underlying buffer without writing the GEMM result, that would explain the zero output (uninitialized GPU memory).
3. Possible quick workaround at the DenseLayer level: when activation is null, route to a plain `Engine.Linear` / `Engine.MatMul + AddBias` call instead of `FusedLinearGpu` — but the real fix probably belongs inside `FusedLinearGpu` itself.

## Relation to #1625
- #1625 was the use-after-dispose on the returned tensor (`using var gpuResult = ForwardGpu(input)`).
- That bug masked this one: pre-fix, EVERY GPU Predict returned zeros because of the disposal; post-fix, only the linear-activation Predict returns zeros because of this bug. The fix for #1625 is correct and complete for what it does; this is a separate problem.

## Environment
- AiDotNet branch: `fix/predict-returns-all-zero` (PR #1628) at HEAD
- AiDotNet.Tensors: 0.96.1 (from NuGet)
- DirectGpuTensorEngine auto-detected on Windows 11
- .NET 10.0

**Comment by ooples** [comment]
## Root cause re-diagnosed — bug is in DenseLayer, not FusedLinearGpu

Tracing inside `FusedLinearGpu` with explicit logging showed every "linear regression" call was actually running with `activation=ReLU`, not `activation=None`. Backtracking:

`DenseLayer.cs:335`:
```csharp
public DenseLayer(int outputSize, IActivationFunction<T>? activationFunction = null, ...)
    : base(new[] { -1 }, new[] { outputSize }, activationFunction ?? new ReLUActivation<T>())
```

When the caller omits `activationFunction` or passes `null`, the constructor silently substitutes `new ReLUActivation<T>()`. So `new DenseLayer<double>(outputSize: 1)` (with a `// linear output for regression` comment in the facade walkthrough's Example 2) actually instantiates a Dense layer with **ReLU output**.

After training:
- If the model's pre-activation goes negative for the test set, ReLU clamps every output to exactly 0.0
- The gradient through dead-ReLU is also 0, so the model can't recover via further training
- Result: regression models with `new DenseLayer<T>(outputSize: 1)` as the output head get stuck at all-zero predictions

## Verified

Forced `IdentityActivation` explicitly:
```csharp
new DenseLayer<double>(outputSize: 1, activationFunction: new IdentityActivation<double>())
```
→ pred = `(-0.05, 0.02, 0.04, 0.01, -0.06)` (varied, non-zero). `FusedLinearGpu` correctly hits its `FusedActivationType.None` path and returns real `input @ W + b` values. **`FusedLinearGpu` is not broken — apologies for the original mis-diagnosis.**

## Fix

PR #1628 commit [bc6c86521](https://github.com/ooples/AiDotNet/commit/bc6c86521) changes the `DenseLayer` default from `?? new ReLUActivation<T>()` to `?? new IdentityActivation<T>()`. Matches Keras / PyTorch `nn.Linear` / TF `Dense` conventions.

## Follow-up scope

10 other Layer types (`ConvolutionalLayer`, `FullyConnectedLayer`, `FeedForwardLayer`, `LocallyConnectedLayer`, `DeconvolutionalLayer`, `LambdaLayer`, `MeshEdgeConvLayer`, `SparseLinearLayer`, `SpiralConvLayer`, `TimeDistributedLayer`) have the same `?? new ReLUActivation<T>()` pattern. They're rarely used as output layers (where dying-ReLU bites), so the user-visible impact is lower, but for API consistency they should follow the same convention. Not in this PR's scope — should be a separate sweep.
